pkg/controller/baremetalhost/host_state_machine.go

```diff
@@ -87,6 +87,24 @@ func (hsm *hostStateMachine) updateHostStateFrom(initialState metal3v1alpha1.Pro
 			stateChanges.With(stateChangeMetricLabels(initialState, hsm.NextState)).Inc()
 		})
 		hsm.Host.Status.Provisioning.State = hsm.NextState
+		// Here we assume that if we're being asked to change the
+		// state, the return value of ReconcileState (our caller) is
+		// set up to ensure the change in the host is written back to
+		// the API. That means we can safely update any status fields
+		// along with the state.
+		switch hsm.NextState {
+		case metal3v1alpha1.StateRegistering,
+			metal3v1alpha1.StateInspecting,
+			metal3v1alpha1.StateProvisioning:
+			// TODO: When the user-selectable profile field is
+			// removed, move saveHostProvisioningSettings() from the
+			// controller to this point. We can't move it yet because
+			// it needs error handling logic that we can't support in
+			// this function.
+			if updateBootModeStatus(hsm.Host) {
+				info.log.Info("saving boot mode")
```

Comment:
Not that we have to fix it in this PR, but this log output would potentially be more useful if it included the bootMode that we saved?

Reply:
Sure, that makes sense. Done.